src/app/components/client/toolbar/UpsellBadge.tsx

```diff
@@ -27,19 +26,14 @@ import { ExperimentData } from "../../../../telemetry/generated/nimbus/experimen
 import { FeatureFlagName } from "../../../../db/tables/featureFlags";
 import { WaitlistDialog } from "../SubscriberWaitlistDialog";
 
-export type UpsellButtonProps = {
-  monthlySubscriptionUrl: string;
-  yearlySubscriptionUrl: string;
-  subscriptionBillingAmount: {
-    yearly: number;
-    monthly: number;
-  };
+export type UpsellLinkButtonProps = {
   enabledFeatureFlags: FeatureFlagName[];
 };
 
-export function UpsellButton(
-  props: UpsellButtonProps & {
-    label: string;
+export function UpsellLinkButton(
+  props: UpsellLinkButtonProps & {
+    children: ReactNode;
+    className?: string;
```

Comment:
Question: why not add these directly to `UpsellLinkButtonProps`?

I'm also not a fan of exposing `className` to parent components: it requires the parent to be aware of implementation details of the child, and in turn the child has a harder time safely making styling changes, because it will have to check every call site to see how it will effect those styles. And I think in this case it's not being used anywhere?

Reply:
`<UpsellLinkButton>` is used by `<UpsellToggleLinkButton>` and ultimately `<UpsellBadge>`, but `<UpsellLinkButton>` is also used standalone in other places. `UpsellLinkButtonProps` is reused for `<UpsellBadge>` that should not supposed to accept these props.